### PR TITLE
Move the marbles if the player has entered the move and it is legal

### PR DESCRIPTION
**Pull request by arimet**
Move the marbles if the player has entered the move and it is legal.

Branch created from 'move-marble'
Need other PR : https://github.com/marmelab/kuba-integration/pull/7 and https://github.com/marmelab/kuba-integration/pull/6

Peek 2021-09-01 17-13

### CODE REVIEW

**Review by Aherbeth — APPROVED** [review]

File: src/Board.ts [hunk 1]
```diff
@@ -112,3 +120,34 @@ function hasFreeSpotBeforeToMove(
 
   return !positionExistsInBoard(boardGraph, `${hIndex},${vIndex}`);
 }
+
+
+export function moveMarble(
+  board: Board,
+  userMove: { marblePosition: string; marbleDirection: string }
```
Comment (Clebiez) [inline src/Board.ts:127]: marbleDirection is a char with only 4 possibilities, Z is not allowed, you can use enum to be more explicit

https://stackoverflow.com/questions/50248807/what-is-the-typescript-equivalent-of-proptypes-oneof-restrict-a-variable-to-s/50248842

File: src/UserInput.ts [hunk 2]
```diff
@@ -17,10 +17,11 @@ export function askUserBoardPath(): Promise<string> {
   });
 }
 
-export async function askUserMove(): Promise<void> {
-  const userMable = await marble();
-  const userDirection = await direction();
+export async function askUserMove(): Promise<{marblePosition: string, marbleDirection: string}> {
```
Comment (Clebiez) [inline src/UserInput.ts:20]: Just to be sure, how the system works if your user enter a non authorized position ?
I saw above a console log if this user can't move but can he try again another input ?

Comment (Clebiez) [reply src/UserInput.ts:20]: Is it the role to readline ? It seems to be a little magical for me, maybe in this case add a little comment to explain that.

**Review by Clebiez — CHANGES_REQUESTED** [review]

File: src/Board.ts (hunk 1 above)
Comment (arimet) [reply src/Board.ts]: Alexandre created a Direction type in another development, I will apply it. Do you think it's enough?

File: src/UserInput.ts (hunk 2 above)
Comment (arimet) [reply src/UserInput.ts:22]: Handling errors and asking the user to restart the move will be handled in my next PR.

I am adding a comment for readline, can you tell me if the comment format is ok?

**Review by Clebiez — APPROVED** [review]

**Pull request merged**